frontend/src/components/navigation/main-sidebar.tsx

```diff
@@ -16,30 +16,11 @@ import {
 } from "lucide-react";
 import { UserMenu } from "@/components/navigation/user-menu.tsx";
 import { useApplicationContext } from "@/lib/use-application-context.ts";
-import {useEffect} from "react";
-import {ApiClient} from "@/lib/api.ts";
-import {authenticationProviderInstance} from "@/lib/authentication-provider.ts";
 
 export function MainSidebar() {
   const { account, dataSetId } = useApplicationContext()!;
-  const { isLoadingAgents, setIsLoadingAgents } = useApplicationContext()!;
-  const { availableAgents, setAvailableAgents } = useApplicationContext()!;
-
-  useEffect(() => {
-    const fetchAgents = async () => {
-      try {
-        const apiClient = new ApiClient(authenticationProviderInstance);
-        const agents = await apiClient.agents().getAvailableAgents();
-        setAvailableAgents(agents);
-      } catch (error) {
-        console.error('Failed to fetch available agents:', error);
-      } finally {
-        setIsLoadingAgents(false);
-      }
-    };
-
-    fetchAgents();
-  }, []);
+  const { isLoadingAgents } = useApplicationContext()!;
+  const { availableAgentInstances } = useApplicationContext()!;
```

Comment:
And this one - I'd just call it availableAgents. 
If we're then asking for a list of possible types of agents in another place, I'd call it availableAgentTypes.